### PR TITLE
✨ Update bootstrap-kubestellar.sh

### PR DESCRIPTION
**Pull request by francostellari**
## Summary

Fix the issue retrieving the latest release of kcp and kubestellar

Fixes #
#760
#767

### CODE REVIEW

**Comment by francostellari** [comment]
@dumb0002

File: bootstrap/bootstrap-kubestellar.sh [hunk 1]
```diff
@@ -76,7 +76,7 @@ kcp_version() {
 }
 
 kcp_get_latest_version() {
-    curl -sL https://github.com/kcp-dev/kcp/releases/latest | grep "</h1>" | head -n 1 | sed -e 's/<[^>]*>//g' | xargs
+    curl -sL https://github.com/kcp-dev/kcp/releases/latest | grep "</h1>" | tail -n 1 | sed -e 's/<[^>]*>//g' | xargs
```
Comment (MikeSpreitzer) [inline bootstrap/bootstrap-kubestellar.sh:79]: This is not wrong, but I wonder if you would rather have a more robust search based on more details of the content (e.g., `fgrep '<h1 data-view-component="true"'`).  Following is what I see right now.

```console
(base) mspreitz@mjs12 kubestellar % curl -sL https://github.com/kcp-dev/kcp/releases/latest | grep "</h1>"
      <h1 id="search-suggestions-dialog-header" class="sr-only">Search code, repositories, users, issues, pull requests...</h1>
      </h1>
      </h1>
        <h1 data-view-component="true" class="d-inline mr-3">v0.20.0</h1>
```

Comment (francostellari) [reply bootstrap/bootstrap-kubestellar.sh:79]: @MikeSpreitzer ideally, I would like to have a file `latest.txt` and `stable.txt` in the repo pointing to the stable and latest release.

See what go does: `curl 'https://go.dev/VERSION?m=text'`

Comment (MikeSpreitzer) [reply bootstrap/bootstrap-kubestellar.sh:79]: OTOH, including in a commit a file that references the commit is problematic.  That is why I think that a GitHub API would be the best solution.

Comment (MikeSpreitzer) [reply bootstrap/bootstrap-kubestellar.sh:79]: If we had a different repo in which we could put the pointer to "stable" and "latest", that would be a resolution.

Comment (MikeSpreitzer) [reply bootstrap/bootstrap-kubestellar.sh:79]: How do the go developer control what `curl 'https://go.dev/VERSION?m=text'` produces?

Comment (MikeSpreitzer) [reply bootstrap/bootstrap-kubestellar.sh:79]: Also, one of the answers in https://stackoverflow.com/questions/51933636/url-for-latest-stable-go-release says that `https://go.dev/dl/?mode=json` should be read instead.

Comment (francostellari) [reply bootstrap/bootstrap-kubestellar.sh:79]: > OTOH, including in a commit a file that references that commit (or a tag that will be created after the commit exists) is problematic. That is why I think that a GitHub API would be the best solution.

@MikeSpreitzer I don't share your concern. The main branch will hold the stable/latest files, not the tag releases. The stable could be rolled back anytime if  an issue is discovered.
After the binary release is done, the main branch will decide to update the files as appropriate to advance the users.
The url to stable/latest.txt should be immutable, so it should be in the main branch.

**Comment by francostellari** [comment]
I have opened https://github.com/kubestellar/kubestellar/pull/763

**Comment by francostellari** [comment]
@clubanderson This PR should fix your issue #767, which is the same issue reported in my #760

**Comment by clubanderson** [comment]
ok, so we are waiting on you and @MikeSpreitzer to hash this out.  Thank you

**Comment by francostellari** [comment]
> ok, so we are waiting on you and @MikeSpreitzer to hash this out. Thank you

**Comment by francostellari** [comment]
Sorry I closed by mistake

**Comment by francostellari** [comment]
Since @MikeSpreitzer agrees that this PR fixes the problem, I recommend pushing the fix ASAP.
I'm using the VERSION file in PR #763 to have a long term fix for KubeStellar, alas not kcp

**Comment by clubanderson** [comment]
our QS is broken, this should fix it, convo can continue in another PR at a later time.

/lgtm
/approve

**Pull request merged**